src/java.base/share/classes/sun/text/CompactByteArray.java

```diff
@@ -77,12 +77,13 @@ public CompactByteArray(byte defaultValue)
         values = new byte[UNICODECOUNT];
         indices = new short[INDEXCOUNT];
         hashes = new int[INDEXCOUNT];
-        for (i = 0; i < UNICODECOUNT; ++i) {
-            values[i] = defaultValue;
+        if (defaultValue != (byte)0) {
+            for (i = 0; i < UNICODECOUNT; ++i) {
+                values[i] = defaultValue;
+            }
```

Comment:
should this be `Arrays.fill(values, defaultValue);` ?

https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/Arrays.html#fill(byte%5B%5D,byte)